pkg/client/introduce/client_test.go

```diff
@@ -53,106 +56,250 @@ func TestNew(t *testing.T) {
+	storageProvider.EXPECT().OpenStore(introduce.Introduce).Return(store, nil)
+
+	provider := mocks.NewMockProvider(ctrl)
 	provider.EXPECT().Service(introduce.Introduce).Return(DIDComm, nil)
+	provider.EXPECT().StorageProvider().Return(storageProvider)
+
+	client, err := New(provider)
+	require.NoError(t, err)
+
+	client.newUUID = func() string { return UUID }
+	require.NoError(t, client.HandleRequestWithInvitation(opts.Inv, opts.Dests[0]))
+}
+
+func TestClient_ProtocolDependency(t *testing.T) {
+	const UUID = "382a7cf8-2c57-4f2f-9359-8ac45b7b4b1f"
+
+	svc, err := introduce.New(&protocol.MockProvider{})
+	require.NoError(t, err)
+	require.NotNil(t, svc)
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	opts := options{
+		Inv: &didexchange.Invitation{
+			ID: UUID,
+		},
+		Dests: []*service.Destination{{
+			ServiceEndpoint: "service/endpoint",
+		}},
+	}
+
+	store := storageMocks.NewMockStore(ctrl)
+	store.EXPECT().Get("client_options_"+UUID).Return(toBytes(t, opts), nil).Times(1)
+	store.EXPECT().Get("client_options_"+UUID).Return(nil, errors.New("error")).Times(1)
+	store.EXPECT().Get("client_options_"+UUID).Return([]byte(`[]`), nil).Times(1)
 
 	storageProvider := storageMocks.NewMockProvider(ctrl)
-	storageProvider.EXPECT().OpenStore(introduce.Introduce).Return(storageMocks.NewMockStore(ctrl), nil)
+	storageProvider.EXPECT().OpenStore(introduce.Introduce).Return(store, nil)
+
+	provider := mocks.NewMockProvider(ctrl)
+	provider.EXPECT().Service(introduce.Introduce).Return(serviceMocks.NewMockDIDComm(ctrl), nil)
 	provider.EXPECT().StorageProvider().Return(storageProvider)
+
 	client, err := New(provider)
 	require.NoError(t, err)
-	require.NoError(t, client.HandleRequestWithInvitation(nil, nil))
+
+	client.newUUID = func() string { return UUID }
+	dependency := client.ProtocolDependency(UUID)
+	require.Equal(t, opts.Inv, dependency.Invitation())
+	require.Equal(t, opts.Dests, dependency.Destinations())
+
+	// with error
+	dependency = client.ProtocolDependency(UUID)
+	require.Nil(t, dependency.Invitation())
+	require.Nil(t, dependency.Destinations())
+
+	// with error
+	dependency = client.ProtocolDependency(UUID)
+	require.Nil(t, dependency.Invitation())
+	require.Nil(t, dependency.Destinations())
 }
 
 func TestClient_SendRequest(t *testing.T) {
+	const UUID = "382a7cf8-2c57-4f2f-9359-8ac45b7b4b1f"
+
 	svc, err := introduce.New(&protocol.MockProvider{})
 	require.NoError(t, err)
 	require.NotNil(t, svc)
 
 	ctrl := gomock.NewController(t)
 	defer ctrl.Finish()
 
-	provider := mocks.NewMockProvider(ctrl)
 	DIDComm := serviceMocks.NewMockDIDComm(ctrl)
 	DIDComm.EXPECT().HandleOutbound(gomock.Any(), gomock.Any()).Return(nil)
-	provider.EXPECT().Service(introduce.Introduce).Return(DIDComm, nil)
+
+	opts := options{
+		Dests: []*service.Destination{{
+			ServiceEndpoint: "service/endpoint",
+		}},
+	}
+
+	store := storageMocks.NewMockStore(ctrl)
+	store.EXPECT().Put("client_options_"+UUID, toBytes(t, opts)).Return(nil).Times(1)
```

Comment:
I see the same string literal used here.. using the constant is beneficial for easier referencing.